blueprints/api.guidelines.md

```diff
@@ -0,0 +1,535 @@
+request body was improperly encoded. In most cases, this is also the proper
+response code when the client submits data that is not semantically correct. For
+example, dates/times represented as timestamp strings in an unsupported format should trigger this response code.
+
+This is also the default "client failure" response code for any other request.
+
+The response body MUST include an entry in the `alerts` array that describes to
+the client what was wrong with the request.
+
+#### 401 Unauthorized
+This MUST be the response code when a client without valid authorization
+information in the HTTP headers requests a resource which cannot be accessed
+without first authorizing. Which should be everything except `/ping` and
+endpoints that provide authorization.
+
+#### 403 Forbidden
+This MUST be used whenever the client is logged-in, but still does not have
+access to the resource they are requesting. It MUST also be used when they have
+some access to the resource, but not with the specific request method they used.
+This can pertain to restricted access on the basis of Role, User Permissions, as
+well as Tenancy.
+
+The response body MUST NOT disclose any information regarding why the user was
+denied access.
+
+#### 404 Not Found
+This MUST be the returned status code when the client requests a path that does
+not exist on the server.
+
+#### 409 Conflict
+This SHOULD be used when the request cannot be completed because the current
+state of the server is fundamentally incompatible with the request. For example,
+creating a new user with an email that is already in use should result in this
+response.
+
+Additionally, this MAY be used instead of `404 Not Found` when the client is
+requesting a link between an object identified by the request URI and some
+other object (e.g. when assigning a server to a Delivery Service) when the other
+object does not exist. If the request URI identifies an object that does not
+exist, the response MUST use `404 Not Found` instead.
+
+This is also the proper response status code when the conditions of a request
+cannot be met, e.g. when a client submits a PATCH request for a resource with an
+If-Match header that does not match the stored object's ETag.
+
+The request body MUST indicate what the conflict is that prevented the request
+from being fulfilled via one or more `error`-level alerts.
+
+#### 500 Internal Server Error
+When the Traffic Ops server encounters some error - through no fault of the
+client or their request - that renders it incapable of servicing the client's
+request, it MUST return this status code if no other code is more appropriate.
+The response body in this case SHOULD indicate that an error occurred, but
+MUST NOT divulge details about what data was being processed, what (if any)
+other components are not functioning properly, or what process failed. Generally
+it is advisable that the resultant \code{alerts} array entry just say "Internal
```

Comment:
do you need to state that the details of the error (if known) should be written to TO logs instead?

Reply:
I would hope not. This is just about the API itself.